src/main/java/ee/tuleva/onboarding/mandate/email/MandateEmailService.java

```diff
@@ -26,41 +26,50 @@
   private final EmailService emailService;
   private final MandateEmailContentService emailContentService;
   private final Clock clock;
+  private final MessageSource messageSource;
 
   public void sendMandate(
       User user,
       Mandate mandate,
       PillarSuggestion pillarSuggestion,
       UserPreferences contactDetails,
       Locale locale) {
-    switch (pillarSuggestion.getOtherPillar()) {
-      case 2 -> sendSecondPillarEmail(user, mandate, pillarSuggestion, locale);
-      case 3 -> sendThirdPillarEmails(user, mandate, pillarSuggestion, contactDetails, locale);
-      default -> throw new IllegalArgumentException(
-          "Unknown pillar: " + pillarSuggestion.getOtherPillar());
+    if (mandate.getPillar() == 2) {
+      sendSecondPillarEmail(user, mandate, pillarSuggestion, locale);
+    } else {
+      sendThirdPillarEmails(user, mandate, pillarSuggestion, contactDetails, locale);
     }
   }
 
   private void sendSecondPillarEmail(
       User user, Mandate mandate, PillarSuggestion pillarSuggestion, Locale locale) {
+    String subject = messageSource.getMessage("mandate.email.secondPillar.subject", null, locale);
     MandrillMessage mandrillMessage =
         emailService.newMandrillMessage(
             emailService.getRecipients(user),
-            "Pensionifondi avaldus",
+            subject,
             getSecondPillarContent(user, mandate, pillarSuggestion, locale),
-            getMandateTags(pillarSuggestion),
+            getSecondPillarMandateTags(pillarSuggestion),
             getMandateAttachments(mandate.getSignedFile(), user, mandate.getId()));
     emailService.send(user, mandrillMessage);
   }
 
+  private List<String> getSecondPillarMandateTags(PillarSuggestion pillarSuggestion) {
```

Comment:
Why no tags for the 3rd pillar mandate?

Reply:
They do have tags, but since they're now in separate methods, they didn't need a method to build them.